public/static/app/perf-bb/PerfBBOutliersCtrl.js

```diff
@@ -0,0 +1,269 @@
+        }, {
+            id: 'suspicious',
+            name: 'Suspicious',
+        }, {
+            id: 'marked',
+            name: 'Marked',
+        }, {
+            id: 'muted',
+            name: 'Muted',
+        }],
+        value: 'detected',
+    };
+
+    // Holds currently selected items.
+    vm.selection = [];
+
+    const modeToItemVisibilityMap = {
+        detected: function(item) { return item.type !== OUTLIERS_TYPE.DETECTED },
+        suspicious: function(item) { return item.type !== OUTLIERS_TYPE.SUSPICIOUS },
+    };
+
+    // Could not be overridden by persistent user settings.
+    const mandatoryDefaultFiltering = {
+        create_time: '>' + moment().subtract(2, 'weeks').format(FORMAT.ISO_DATE),
+        project: '=' + $window.project,
+    };
+
+    const getDefaultFiltering = () => {
+        let typeFilter = {};
+        if (vm.mode.value in modeToItemVisibilityMap) {
+            typeFilter = {
+                type: '=' + vm.mode.value,
+            };
+        }
+        return _.extend(
+            {},
+            mandatoryDefaultFiltering,
+            typeFilter
+        );
+    };
+
+    vm.state = {
+        filtering: getDefaultFiltering,
+        lookBackDays: 14,
+        mode: vm.mode.value,
+    };
+
+    const modeToCollMap = {
+        detected: STITCH_CONFIG.PERF.COLL_OUTLIERS,
+        suspicious: STITCH_CONFIG.PERF.COLL_OUTLIERS,
+        muted: STITCH_CONFIG.PERF.COLL_MUTE_OUTLIERS,
+        marked: STITCH_CONFIG.PERF.COLL_MARKED_OUTLIERS,
+    };
+
+    vm.lookBackDays = vm.state.lookBackDays;
+
+    vm.applyFiltering = function() {
+        // Say the form the state is 'pristine'
+        $scope.form.$setPristine();
+        // Update controller state
+        vm.state.lookBackDays = +vm.lookBackDays;
+        loadData();
+    };
+
+    // Enhances filtering state with some contextual meta data
+    // This data is required by expression compiler
+    function getFilteringContext(state) {
+        return _.reduce(state.filtering(), (m, v, k) => {
+            if (!getCol) {
+                return m;
+            }
+            const col = getCol(k);
+            if (!col) return m;  // Error! Associated col does not found
+
+            return m.concat({
+                field: k,
+                term: v,
+                type: col.colDef.type || 'string',
+            })
+        }, []);
+    }
+
+    // Creates aggregation expression, which could be used by Stitch
+    // for given `state`
+    function getAggChain(state) {
+        let chain = [];
+
+        // Check if the state has filtering
+        if (state.filtering) {
+            const filteringChain = MDBQueryAdaptor.compileFiltering(
+                // filtering context enhances state data with important meta data
+                getFilteringContext(state)
+            );
+            // check if filtering query was compiled into something
+            filteringChain && chain.push(filteringChain);
```

Comment:
Checking for null seems unnecessary unless the function you initialize it with can return null/undefined

Reply:
I think there are cases where a null/undefined value would be expected.